flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -131,10 +132,17 @@ public boolean dropTemporarySystemFunction(String name, boolean ignoreIfNotExist
                             "Could not drop temporary system function. A function named '%s' doesn't exist.",
                             name));
         }
+        unregisterFunctionJarResources(function);
```

Comment:
Does hive or spark remove the resources from the ResourceManager? @lsyldliu

Reply:
@fsk119 Currently multiple jobs will share the `ResourceManager` in `Session`, I think it's better to remove the jars for  customized function when they are not used anyway.